tests/upgrades/test_repository.py

```diff
@@ -371,3 +372,151 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
+
+        :id: preupgrade-d6661342-a348-4dd5-9ddf-3fd630b1e286
+
+        :steps:
+            1. Before satellite upgrade
+            2. Create new organization and location
+            3. Upload a manifest in it
+            4. Create a ak with repos and no subscriptions added in it
+            5. Create a content host
+            6. Register the content host
+            7. Confirm content host only has red hat repo enabled
+            8. Confirm content host is in entitlement mode
+
+        :expectedresults:
+            1. Content host is registered in entitlement mode and has only the red hat repository
+            enabled on host.
+
+        """
+        _, manifester = upgrade_entitlement_manifest
+        org = upgrade_entitlement_manifest_org
+        rhel_contenthost._skip_context_checkin = True
+        lce = target_sat.api.LifecycleEnvironment(organization=org).create()
+        product = target_sat.api.Product(organization=org).create()
+        custom_repo = target_sat.api.Repository(
+            content_type='yum', product=product, url=settings.repos.module_stream_1.url
+        ).create()
+        custom_repo.sync()
+        if rhel_contenthost.os_version.major > 7:
+            rh_repo_id = target_sat.api_factory.enable_sync_redhat_repo(
+                constants.REPOS[f'rhel{rhel_contenthost.os_version.major}_bos'], org.id
+            )
+        else:
+            rh_repo_id = target_sat.api_factory.enable_sync_redhat_repo(
+                constants.REPOS[f'rhel{rhel_contenthost.os_version.major}'], org.id
+            )
+        rh_repo = target_sat.api.Repository(id=rh_repo_id).read()
+        assert rh_repo.content_counts['rpm'] >= 1
+        repo_list = [custom_repo, rh_repo]
+        content_view = target_sat.publish_content_view(org, repo_list)
+        content_view.version[-1].promote(data={'environment_ids': lce.id})
+        subscription = target_sat.api.Subscription(organization=org.id).search(
+            query={'search': f'name="{constants.DEFAULT_SUBSCRIPTION_NAME}"'}
+        )
+        assert len(subscription)
+        activation_key = target_sat.api.ActivationKey(
+            content_view=content_view,
+            organization=org,
+            environment=lce,
+        ).create()
+        all_content = activation_key.product_content(data={'content_access_mode_all': '1'})[
+            'results'
+        ]
+        for content in all_content:
+            if content['name'] == custom_repo.name:
+                custom_content_label = content['label']
+            if content['repositories'][0]['id'] == rh_repo_id:
+                rh_content_label = content['label']
+        activation_key.content_override(
+            data={'content_overrides': [{'content_label': custom_content_label, 'value': '1'}]}
+        )
+        rhel_contenthost.install_katello_ca(target_sat)
+        rhel_contenthost.register_contenthost(org=org.name, activation_key=activation_key.name)
```

Comment:
```suggestion
        rhel_contenthost.register(org=org, loc=None, activation_key=activation_key.name, target=target_sat)
```
Let's go with global registration, it should work the same.